cloud/linode/loadbalancers.go

```diff
@@ -316,30 +358,35 @@ func (l *loadbalancers) EnsureLoadBalancerDeleted(ctx context.Context, clusterNa
 	return nil
 }
 
-// The returned error will be errLbNotFound if the load balancer does not exist.
-func (l *loadbalancers) lbByName(ctx context.Context, client *linodego.Client, name string) (*linodego.NodeBalancer, error) {
-	jsonFilter, err := json.Marshal(map[string]string{"label": name})
-	if err != nil {
-		return nil, err
-	}
-	lbs, err := l.client.ListNodeBalancers(ctx, linodego.NewListOptions(0, string(jsonFilter)))
+func (l *loadbalancers) getNodeBalancerByIPv4(ctx context.Context, service *v1.Service, ipv4 string) (*linodego.NodeBalancer, error) {
+	lbs, err := l.client.ListNodeBalancers(ctx, nil)
 	if err != nil {
 		return nil, err
 	}
-
-	if len(lbs) > 0 {
-		return &lbs[0], nil
+	for _, lb := range lbs {
+		if *lb.IPv4 == ipv4 {
+			klog.Infof("found NodeBalancer (%d) for service %s/%s via IPv4 (%s)", lb.ID, service.Namespace, service.Name, ipv4)
+			return &lb, nil
+		}
 	}
-
 	return nil, errLbNotFound
 }
 
-func (l *loadbalancers) createNodeBalancer(ctx context.Context, service *v1.Service, configs []*linodego.NodeBalancerConfigCreateOptions) (*linodego.NodeBalancer, error) {
-	lbName := cloudprovider.GetLoadBalancerName(service)
+func (l *loadbalancers) getNodeBalancerByID(ctx context.Context, service *v1.Service, id int) (lb *linodego.NodeBalancer, err error) {
+	lb, err = l.client.GetNodeBalancer(ctx, id)
+	if err != nil {
+		if apiErr, ok := err.(*linodego.Error); ok && apiErr.Code == http.StatusNotFound {
+			err = errLbNotFound
+		}
+	} else {
+		klog.Infof("found NodeBalancer (%d) for service %s/%s via %s annotation", id, service.Namespace, service.Name, annLinodeNodeBalancerID)
+	}
+	return
```

Comment:
It looks like you can drop the `else` if you return early during the error condition, like so:
```suggestion
	lb, err = l.client.GetNodeBalancer(ctx, id)
	if err != nil {
		if apiErr, ok := err.(*linodego.Error); ok && apiErr.Code == http.StatusNotFound {
			return nil, errLbNotFound
		}
	    return nil, err
	}
	klog.Infof("found NodeBalancer (%d) for service %s/%s via %s annotation", id, service.Namespace, service.Name, annLinodeNodeBalancerID)
	return lib, nil
```